moto/ecs/responses.py

```diff
@@ -71,46 +96,49 @@ def put_cluster_capacity_providers(self) -> str:
         cluster = self.ecs_backend.put_cluster_capacity_providers(
             cluster_name, capacity_providers, default_capacity_provider_strategy
         )
-        return json.dumps({"cluster": cluster.response_object})
+        return ActionResult({"cluster": cluster})
 
-    def delete_capacity_provider(self) -> str:
+    def delete_capacity_provider(self) -> ActionResult:
         name = self._get_param("capacityProvider")
         provider = self.ecs_backend.delete_capacity_provider(name)
-        return json.dumps({"capacityProvider": provider.response_object})
+        return ActionResult({"capacityProvider": provider})
 
-    def update_capacity_provider(self) -> str:
+    def update_capacity_provider(self) -> ActionResult:
         name = self._get_param("name")
         asg_provider = self._get_param("autoScalingGroupProvider")
         provider = self.ecs_backend.update_capacity_provider(name, asg_provider)
-        return json.dumps({"capacityProvider": provider.response_object})
+        return ActionResult({"capacityProvider": provider})
 
-    def describe_capacity_providers(self) -> str:
+    def describe_capacity_providers(self) -> ActionResult:
         names = self._get_param("capacityProviders")
         providers, failures = self.ecs_backend.describe_capacity_providers(names)
-        return json.dumps(
+        return ActionResult(
             {
-                "capacityProviders": [p.response_object for p in providers],
-                "failures": [p.response_object for p in failures],
+                "capacityProviders": providers,
+                "failures": failures,
             }
         )
 
-    def describe_clusters(self) -> str:
-        names = self._get_param("clusters")
-        include = self._get_param("include")
-        clusters, failures = self.ecs_backend.describe_clusters(names, include)
-        return json.dumps(
-            {
-                "clusters": clusters,
-                "failures": [cluster.response_object for cluster in failures],
-            }
-        )
+    def describe_clusters(self) -> ActionResult:
+        cluster_names = self._get_param("clusters", [])
+        clusters, failures = self.ecs_backend.describe_clusters(cluster_names)
+        include = self._get_param("include", [])
+        if "TAGS" in include:
+            self._include_in_response(DESCRIBE_CLUSTERS_TAGS)
+        else:
+            self._exclude_from_response(DESCRIBE_CLUSTERS_TAGS)
```

Comment:
Moto only supports `TAGS` at the moment, but there are lots of other possible values in the `include` parameter.  Supporting these in the future would be quite simple, e.g. having a map of `include` param names with their associated response path, which could simply be looped through to include/exclude as needed per request.